### PR TITLE
Use reusable workflows from web-platform-packages

### PR DESCRIPTION
**Pull request by nayres**
🎟️ [Asana Task]()

---
## Description

This PR enables this repository to use new/ centrally located Canary & Release workflow yaml files. These centrally located workflows are triggered when a pull request is created, and new code is pushed (respectively).

### Known Limitations
These workflows have not been tested locally due to the limitations of GH Actions.

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [x] Add Asana and Preview links above.
- [x] Conduct thorough self-review.
- [x] Add or update tests as appropriate.
- [x] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [x] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [x] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -7,29 +7,7 @@ on:
...
-      - name: Create Release Pull Request or Publish to npm
-        id: changesets
-        uses: changesets/action@master
-        with:
-          # This expects you to have a script called release which does a build for your packages and calls changeset publish
-          publish: npm run release
-        env:
-          GITHUB_TOKEN: ${{ secrets.CHANGESETS_PAT }}
-          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+    uses: hashicorp/web-platform-packages/.github/workflows/release.yml
```
Comment (BRKalow) [inline .github/workflows/release.yml:10]: [question] will this always pull from the `main` branch?

Comment (nayres) [reply .github/workflows/release.yml:10]: > [question] will this always pull from the `main` branch?

Yes, the previous/ existing workflow defines `main` as the branch to pull from.

File: .github/workflows/canary-release.yml [hunk 2]
```diff
@@ -10,60 +10,7 @@ on:
...
-            Latest commit: ${{ github.event.pull_request.head.sha }}
-
-            <details><summary>Published ${{ steps.packagesData.outputs.count }} packages</summary>
-
-            ${{ fromJSON(steps.packagesData.outputs.packageList) }}
-
-            </details>
+    uses: hashicorp/web-platform-packages/.github/workflows/canary-release.yml
+    secrets:
+      PUBLISH_GITHUB_TOKEN: ${{ secrets.PUBLISH_GITHUB_TOKEN }}
```
Comment (BRKalow) [inline .github/workflows/canary-release.yml:15]: Let's use `CHANGESETS_PAT` here to align with the other workflow:
```suggestion
      PUBLISH_GITHUB_TOKEN: ${{ secrets.CHANGESETS_PAT }}
```

**Review by BRKalow — APPROVED** [review]
👏

**Pull request merged**